spec/requests/api/v1/teachers_request_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe 'Teachers API' do
+  it 'sends a list of teachers' do
+    teachers = create_list(:teacher, 3)
+    get '/api/v1/teachers'
+
+    expect(response).to be_successful
+    teachers = JSON.parse(response.body, symbolize_names: true)
+
+    expect(teachers).to have_key(:data)
+    expect(teachers[:data]).to be_an(Array)
+    expect(teachers[:data].count).to eq(3)
+
+    teacher = teachers[:data][0]
+
+    expect(teacher).to have_key(:id)
+    expect(teacher[:id]).to be_an(String)
+
+    expect(teacher).to have_key(:attributes)
+    expect(teacher[:attributes]).to be_a(Hash)
+
+    expect(teacher[:attributes]).to have_key(:name)
+    expect(teacher[:attributes][:name]).to be_a(String)
+
+    expect(teacher).to have_key(:relationships)
+    expect(teacher[:relationships]).to be_a(Hash)
+
+    expect(teacher[:relationships]).to have_key(:courses)
+    expect(teacher[:relationships][:courses]).to be_a(Hash)
+
+    expect(teacher[:relationships][:courses]).to have_key(:data)
+    expect(teacher[:relationships][:courses][:data]).to be_an(Array)
+  end
+
+  it 'can get one teacher by its id' do
+    id = create(:teacher).id
+    get "/api/v1/teachers/#{id}"
+
+    expect(response).to be_successful
+    teacher = JSON.parse(response.body, symbolize_names: true)
+
+    teacher = teacher[:data]
+
+    expect(teacher).to have_key(:id)
+    expect(teacher[:id]).to be_an(String)
+
+    expect(teacher).to have_key(:attributes)
+    expect(teacher[:attributes]).to be_a(Hash)
+
+    expect(teacher[:attributes]).to have_key(:name)
+    expect(teacher[:attributes][:name]).to be_a(String)
+
+    expect(teacher).to have_key(:relationships)
+    expect(teacher[:relationships]).to be_a(Hash)
+
+    expect(teacher[:relationships]).to have_key(:courses)
+    expect(teacher[:relationships][:courses]).to be_a(Hash)
+
+    expect(teacher[:relationships][:courses]).to have_key(:data)
+    expect(teacher[:relationships][:courses][:data]).to be_an(Array)
+  end
+
+  it 'can create a new teacher' do
+    teacher_id = create(:teacher).id
```

Comment:
if we are testing creation of a user, should we be factory creating a teacher here?

Reply:
That's a good point.  I think I remember one of the instructor's advice of making sure not to stub out what you're actually testing.  However, I also think that there is some wiggle room seeing as we are asking our Factory bot to create one for us might be viewed as equivalent to creating one.

Reply:
Sorry, let me clarify. I meant, if the test is for "creating a new teacher", if line 65 necessary. I don't see it being used in the test so I was wondering if it was an accident :)